atomic_defi_design/qml/Dashboard/NotificationsModal.qml

```diff
@@ -264,121 +293,103 @@ BasicModal {
 
                 anchors.fill: parent
                 model: notifications_list
-
-                delegate: Item {
-                    width: list.width
-                    height: 60
-
-                    DefaultText {
-                        anchors.top: parent.top
-                        anchors.topMargin: 10
-                        anchors.right: parent.right
-                        anchors.rightMargin: 200
-                        text_value: modelData.human_date
-                        font.pixelSize: Style.textSizeSmall
+                delegate: ItemDelegate {
+                    function removeNotification() {
+                        notifications_list.splice(index, 1)
+                        notifications_list = notifications_list
                     }
-
-                    ColumnLayout {
-                        anchors.verticalCenter: parent.verticalCenter
-                        anchors.left: parent.left
-                        anchors.leftMargin: 10
-
-                        DefaultText {
-                            text_value: modelData.title
-                            font.pixelSize: Style.textSizeSmall4
-                            font.weight: Font.Medium
-                        }
-
-                        DefaultText {
-                            text_value: modelData.message
-                            font.pixelSize: Style.textSizeSmall1
+                    onClicked: {
+                        // Action might create another event so we save it and then remove the current one, then take the action
+                        performNotificationAction(notifications_list[index])
+                        removeNotification()
+                    }
+                    height: _column.height+10
+                    width: list.width
+                    property int stat: {
+                        if (notifications_list[index].title.indexOf("You received")!==-1 ) {
+                            return 0
+                        } else if(notifications_list[index].title.indexOf("You sent")!==-1) {
+                            return 1
+                        } else {
+                            return 2
```

Comment:
This is not a good way to detect notification, according to the old code, we was able to know which kind of notification are received and how to treat them.

- What means `stat` field ? 
- Later in the code you are using this `stat` field to get others infos such as color or icon, this is a violation of https://en.wikipedia.org/wiki/Don%27t_repeat_yourself (DRY rules) - you can do a single function that will set every information you need once, you can also at the moment you receive the notification set it in a way that retrieving information will be easier such as a map:

`notifications_map[notifications_list[index].kind].color`
`notifications_map[notifications_list[index].kind].icon`

Where `kind` could be :

```cpp
enum notification_kind
{
   send,
   receive,
   others
}
```